librosa/display.py

```diff
@@ -617,6 +618,19 @@ def __envelope(x, hop):
     return x_frame.max(axis=1)
 
 
+_categorical_axis_types = ['chroma', 'chroma_h', 'tonnetz', 'frames']
+_time_axis_types = ['linear', 'fft', 'hz', 'log,' 'fft_note', 'fft_svara', 'mel', 'cqt_hz', 'cqt_note', 'cqt_svara']
+_freq_axis_types = ['time', 's', 'ms', 'lag', 'lag_s', 'lag_ms']
+_tempo_axis_types = ['tempo', 'fourier_tempo']
```

Comment:
I'm not sure if we should treat tempo and fourier_tempo as equivalent.  (Honest question.)  The hiccup here is that `tempo` can include "infinite" bpm while fourier_tempo cannot, and this might do some screwy things if we try to match them.

The more I think about this, I'm starting to wonder if there would be value in doing an allclose check on the actual coordinates used, rather than matching on the shape of the array being displayed.

Reply:
Thanks for the review. I also think dividing the types into smaller groups makes more sense.
I don't get what should be checked with allclose though. Is checking xlim and ylim inadequate?

Reply:
> I don't get what should be checked with allclose though. Is checking xlim and ylim inadequate?

I'm not sure.  Limit-matching is obviously necessary, but it might not be sufficient for imposing an equal aspect ratio.  We can have time-vs-time over the same limits but with different spacing along one axis (while still having the same array shape).  This is admittedly a very unusual use-case, but I just want to make sure we think through the implications here.

Reply:
Do you mean doing something like `np.allclose(x_coords, y_coords)` instead of `xlim == ylim`?

Reply:
That's what I was thinking, yes.  Like I said, I'm not 100% sure that it's necessary.  Maybe @lostanlen has some thoughts on this?